pkgs/applications/science/meteorology/supercell-wx/default.nix

```diff
@@ -0,0 +1,158 @@
+  glew,
+  geos,
+  git,
+  boost185,
+  spdlog,
+  stb,
+  libcpr_1_10_5,
+  libpng,
+  libSM,
+  geographiclib,
+  howard-hinnant-date,
+  re2,
+  gtest,
+  glm,
+  qtbase,
+  qttools,
+  qtmultimedia,
+  qtpositioning,
+  qtimageformats,
+  tbb_2021_11,
+  tzdata,
+  substituteAll,
+  python3,
+  wrapQtAppsHook,
+}:
+let
+  version = "0.4.4";
+  src = fetchFromGitHub {
+    owner = "dpaulat";
+    repo = "supercell-wx";
+    rev = "refs/tags/v${version}-release";
+    sha256 = "sha256-HghyRFLKboztQEOsvN2VaSxxcuPPu1GIakBNWRIWBOk=";
+    fetchSubmodules = true;
+  };
+
+  gtestSkip = [
+    # Skip tests requiring network access
+    "AwsLevel*DataProvider.FindKeyNow"
+    "AwsLevel*DataProvider.FindKeyFixed"
+    "AwsLevel*DataProvider.LoadObjectByKey"
+    "AwsLevel*DataProvider.Refresh"
+    "AwsLevel*DataProvider.GetAvailableProducts"
+    "AwsLevel*DataProvider.GetTimePointsByDate"
+    "AwsLevel*DataProvider.Prune"
+    "UpdateManagerTest.CheckForUpdates"
+    "WarningsProvider*\"https"
+
+    # These tests are failing (seemingly can't overwrite a file created by earlier test).
+    "SettingsManager/DefaultSettingsTest*"
+    "SettingsManager/BadSettingsTest*"
+  ];
+in
+stdenv.mkDerivation {
+  name = "supercell-wx";
+  inherit version;
+  inherit src;
+
+  meta = {
+    homepage = "https://supercell-wx.rtfd.io";
+    downloadPage = "https://github.com/dpaulat/supercell-wx/releases";
+    description = "Provides live visualization of NEXRAD weather data and alerts.";
+    longDescription = ''
+      Supercell Wx is a free, open source application to visualize live and
+      archive NEXRAD Level 2 and Level 3 data, and severe weather alerts.
+      It displays continuously updating weather data on top of a responsive
+      map, providing the capability to monitor weather events using
+      reflectivity, velocity, and other products.
+    '';
+    license = lib.licenses.mit;
+    mainProgram = "supercell-wx";
+    platforms = [ "x86_64-linux" ];
+    maintainers = with lib.maintainers; [ aware70 ];
+  };
+
+  CXXFLAGS = "-Wno-error=restrict -Wno-error=maybe-uninitialized -Wno-error=deprecated-declarations";
+  env.GTEST_FILTER = "-${lib.concatStringsSep ":" gtestSkip}";
+
+  doCheck = true;
+
+  # These tests aren't built by 'all', but ctest still tries to run them.
+  cmakeFlags = [
+    "-DCMAKE_CTEST_ARGUMENTS=-E;'test_mln_core|test_mln_widgets'"
+  ];
+
+  patches = [
+    # These are for Nix compatibility {{{
+    ./patches/use-find-package.patch # Replace some vendored dependencies with Nix provided versions
+    (substituteAll {
+      # Skip tagging build with git version, and substitute it with the src revision (still uses current year timestamp)
+      src = ./patches/skip-git-versioning.patch;
+      rev = src.rev;
+    })
+    # Prevents using some Qt scripts that seemed to break the install step. Fixes missing link to some targets.
+    ./patches/fix-cmake-install.patch
+    # }}}
+
+    # These may be submitted upstream {{{
```

Comment:
Since once of these was accepted upstream, you can replace it with a fetchpatch of the relevant commit

Reply:
Added fetchpatch for the audio codec fix.